### DISCUSSION TITLE
line_thickness and show_labels arguements don't seem to work.

### DISCUSSION THREAD

**Discussion by wb666greene** [ultralytics/ultralytics#2032] [Q&A] [answered]
I've integrated yolo8 into my security system and it is working very well.  Problem is on the annotated images the thickness of the lines drawing the boxes and the object labels are too large and obscure needed details.

basically I do:
results = model.predict(image, conf=confidence-0.001, line_thickness=1,verbose=False,show_labels=False)
 annotated_image = results[0].plot(line_thickness=1)

The conf= and verbose= do what i expected, while the show_labels= and line_thickness= seem to have no effect.

Any clues as to what I am misunderstanding or doing wrong here?

**Comment by wb666greene** [comment, marked as answer]
This part of the documentation seems incorrect in places.  Playing around I got it to work with:
    
    results = model.predict(image, conf=confidence-0.001, verbose=False)
      annotated_image = results[0].plot(line_width=1, labels=False)

**Reply by glenn-jocher** [reply]
@wb666greene hello! I'm glad to hear that YOLOv8 is working well with your security system. It looks like you've found a solution to your issue with the `line_thickness` and `show_labels` arguments. Indeed, the correct arguments to use within the `plot()` method are `line_width` for adjusting the thickness of the bounding box lines and `labels` to toggle the visibility of the labels.

The `predict()` method is used for running inference, and it doesn't directly control the visualization parameters like `line_thickness` or `show_labels`. These parameters are part of the `plot()` method, which is why your second snippet of code works as expected.

For future reference, you can always check the [Ultralytics Docs](https://docs.ultralytics.com) for the most up-to-date information on the usage of different methods and their parameters. If you find any discrepancies in the documentation, please feel free to raise an issue on the GitHub repository so we can correct it. Your contributions help improve the experience for the entire YOLO community. 😊

If you have any more questions or run into further issues, don't hesitate to reach out. Happy detecting! 🚀

**Comment by dh031200** [comment]
@wb666greene 

When I use `save=True` it works correctly, but when I `plot()` no arguments are passed.
```python
model.predict(verbose=False, show_labels=False, line_width=1, save=True)
```
I added code to make the resulting image available even when the python program is running.

## Before
```python
results = model.predict(verbose=False)
annotated_image = results[0].plot(labels=False, line_width=1)
cv2.imwrite('output.png', annotated_image)
```

## After
```python
results = model.predict(verbose=False, show_labels=False, line_width=1)
annotated_image = results[0].plot()
cv2.imwrite('output.png', annotated_image)
```

If the code is merged, you'll be able to use it the way you thought. #2373
